pkg/utils/ocm.go

```diff
@@ -432,28 +432,56 @@ func GetHiveCluster(clusterId string) (*cmv1.Cluster, error) {
 	return resp.Items().Get(0), nil
 }
 
-// GetManagementCluster returns the OCM Cluster object for a provided clusterId
-func GetManagementCluster(clusterId string) (*cmv1.Cluster, error) {
+// GetMgmtServiceCluster returns the Mgmt Cluster and Service Cluster object for a provided HCP clusterId
+func GetMgmtServiceCluster(clusterId string) (*cmv1.Cluster, *cmv1.Cluster, error) {
 	conn, err := CreateConnection()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	defer conn.Close()
 
+	var svcClusterName, mgmtClusterName string
+
 	hypershiftResp, err := conn.ClustersMgmt().V1().Clusters().
 		Cluster(clusterId).
 		Hypershift().
 		Get().
 		Send()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	if err == nil && hypershiftResp != nil {
+		mgmtClusterName = hypershiftResp.Body().ManagementCluster()
+	}
+
+	if mgmtClusterName == "" {
+		return nil, nil, fmt.Errorf("failed to lookup service cluster for cluster %s", clusterId)
```

Comment:
```suggestion
		return nil, nil, fmt.Errorf("failed to lookup management cluster for cluster %s", clusterId)
```
Should it not be management cluster here?